feature_engine/discretisation/binariser.py

```diff
@@ -0,0 +1,231 @@
+
+    .. [2] Dong. "Beating Kaggle the easy way". Master Thesis.
+        https://www.ke.tu-darmstadt.de/lehre/arbeiten/studien/2015/Dong_Ying.pdf
+
+    Examples
+    --------
+
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> from feature_engine.discretisation import EqualWidthDiscretiser
+    >>> np.random.seed(42)
+    >>> X = pd.DataFrame(dict(x = np.random.randint(1,100, 100)))
+    >>> transformer = BinaryDiscretiser(threshold=50)
+    >>> transformer.fit(X)
+    >>> transformer.transform(X)['x'].value_counts()
+        x
+        1    56
+        0    44
+        Name: count, dtype: int64
+    """
+
+    def __init__(
+        self,
+        threshold: Union[None, int, float] = None,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        return_object: bool = False,
+        return_boundaries: bool = False,
+        precision: int = 3,
+    ) -> None:
+
+        if threshold is None:
+            raise TypeError(
+                "threshold not supplied."
+                " Please provide a threshold of type float or int."
+            )
+
+        if not isinstance(threshold, (int, float)):
+            raise TypeError(
+                "threshold must be an integer or a float."
+                f" Got type '{type(threshold).__name__}' instead."
+            )
+
+        super().__init__(return_object, return_boundaries, precision)
+
+        self.variables = _check_variables_input_value(variables)
+        self.threshold = threshold
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        Learn the boundaries of the bins for each
+        variable.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training dataset. Can be the entire dataframe, not just the variables
+            to be transformed.
+        y: None
+            y is not needed in this encoder. You can pass y or None.
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+
+        failed_threshold_check = []
+        self.binner_dict_ = {}
+        for var in self.variables_:
+            # Check that threshold is within range
+            if (self.threshold < min(X[var])) or (self.threshold > max(X[var])):
+                # Omit these features from transformation step
+                failed_threshold_check.append(var)
+            else:
+                self.binner_dict_[var] = [
+                    float("-inf"),
+                    np.float64(self.threshold),
+                    float("inf"),
+                ]
+
+        if failed_threshold_check:
+            print(
+                "threshold outside of range for one or more variables."
+                f" Features {failed_threshold_check} have not been transformed."
+            )
+
+        # A list of features that satisfy threshold check and will be transformed
+        self.variables_trans_ = [
```

Comment:
we normally store the variables that will be transformed in variables_ But I am not convinced that we need to exclude the variables that will not be transformed. This is something for the user to take care of, not feature-engine.

We can raise a warning, but that is as far as I would go.